packages/lib/shim-init-node.ts

```diff
@@ -714,6 +715,29 @@ function shimInit(options: ShimInitOptions = null) {
 		}
 	};
 
+	shim.pdfExtractEmbeddedText = async (pdfPath: string): Promise<string[]> => {
+		const loadingTask = pdfJs.getDocument(pdfPath);
+		const doc = await loadingTask.promise;
+
+		const pageTasks = [];
+
+		for (let pageNum = 1; pageNum <= doc.numPages; pageNum++) {
+			pageTasks.push((async () => {
+				const page = await doc.getPage(pageNum);
+				const textContent = await page.getTextContent();
+
+				const strings = textContent.items.map(item => {
+					const text = (item as TextItem).str ?? '';
+					return text;
+				}).join('\n');
+
+				return strings;
+			})());
+		}
+
+		return Promise.all(pageTasks);
```

Comment:
My tests were with several hundred page documents.

My reasoning for doing all tasks in parallel is that PDFJs uses a `Worker` internally and I was hopeful that this would decrease overhead from waiting for `postMessage`s to be responded to.

> I'm wondering if pdfjs works well with multiple pages being processed in parallel, and whether they could be performance issues?

It seems to work well, but that could certainly change in the future! I'll change this from `Promise.all` into something sequential.

Reply:
I've applied the change and re-tested.

Reply:
Thanks for checking. I'm never quite sure when it makes sense to use `Promise.all` because it depends on what the called code does and probably how V8 runs all this.